### PR TITLE
Add UserMerge command

### LINKED ISSUE #23: User-merge command
- `Lmc\Matej\Model\Command\UserMerge`
- Static named constructor with helpful name (so you are sure which user is source and which target). Something like `UserMerge::toTargetUserMergeSourceUser($targetUser, $sourceUser)` :-).

Once the command is done, in separate commit (or in separate pull request) the EventsRequestBuilder should be extended by `addUserMerge` and `addUserMerges` methods.

### PR DESCRIPTION
**Pull request by foglcz**
First half of #23

This does not implement any actual call to Matej endpoints - those will be implemented in separate PRs.

### CODE REVIEW

File: src/Model/Command/UserMerge.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+class UserMerge extends AbstractCommand
+{
+    /** @var string */
+    private $sourceUserId;
+    /** @var string */
+    private $targetUserId;
+
+    /**
+     * Merge source user to target user AND DELETE SOURCE USER.
```
Comment (jirinovak) [inline src/Model/Command/UserMerge.php]: Same as in #27. I think we should merge constructor and class comments in this particular case.

File: src/Model/Command/UserMerge.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public function __construct(string $sourceUserId, string $targetUserId)
```
Comment (OndraM) [inline src/Model/Command/UserMerge.php]: All other Commands (ok, both of them :-D) could be instantiated only through the static constructor. My idea was to not mix two approaches (new vs static constructor) and even though some commands do need just one way of instantiation (thus the `::create()` method), I would prefer to make the static constructors the only way of Command instantiation (and make the `__construct()` private), so there is less cognitive load for the developer and also the usage is unified.

File: src/Model/Command/UserMerge.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        return [
+            'target_user_id' => $this->targetUserId,
+            'source_user_id' => $this->sourceUserId,
+        ];
+    }
+
+    /**
+     * Merge source user to target user AND DELETE SOURCE USER.
+     */
+    public static function create(string $sourceUserId, string $targetUserId): self
```
Comment (OndraM) [inline src/Model/Command/UserMerge.php]: I'm just thinking if having both of those constructors (which differs just with parameter order) will not make more mess... 🤔

Comment (OndraM) [inline src/Model/Command/UserMerge.php]: I tend to place the static named constructors right after __construct, so they are "together" on one place.

File: src/Model/Command/UserMerge.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        return new static($sourceUserId, $targetUserId);
+    }
+
+    /**
+     * Merge source user into target user AND DELETE SOURCE USER.
+     */
+    public static function mergeInto(string $targetUserId, string $sourceUserId): self
```
Comment (OndraM) [inline src/Model/Command/UserMerge.php]: Maybe make the "delete" thing more obvious by naming the variable eg. `$sourceUserIdToBeDeleted` :)?

File: tests/Model/Command/UserMergeTest.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\Matej\Model\Command;
+
+use PHPUnit\Framework\TestCase;
+
+class UserMergeTest extends TestCase
+{
+    /**
+     * @test
```
Comment (OndraM) [inline tests/Model/Command/UserMergeTest.php]: I prefer to use one-line annotations if it is possible.

**Review by OndraM — CHANGES_REQUESTED** [review]

File: tests/Model/Command/UserMergeTest.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\Matej\Model\Command;
+
+use PHPUnit\Framework\TestCase;
+
+class UserMergeTest extends TestCase
```
Comment (OndraM) [inline tests/Model/Command/UserMergeTest.php]: We extend "our" TestCase in other tests... It has no added value here right now, but we may do it here as well to make it unified.

File: src/Model/Command/UserMerge.php (hunk 1 above)
Comment (OndraM) [reply src/Model/Command/UserMerge.php]: So you'd like to have the PHPDoc as class comment or as constructor comment?

IMHO we may have some more detailed information and documentation in the class commend, and basic information in the constructor comment.

**Review by OndraM — DISMISSED** [review]

**Review by jirinovak — DISMISSED** [review]

**Pull request merged**